src/main/scala/com/gu/zuora/ProductSubscriptionBuilders.scala

```diff
@@ -0,0 +1,68 @@
+package com.gu.zuora
+
+import com.gu.i18n.Country
+import com.gu.support.catalog.ProductRatePlanId
+import com.gu.support.config.ZuoraConfig
+import com.gu.support.promotions.{PromoCode, PromotionService}
+import com.gu.support.workers.{Contribution, DigitalPack}
+import com.gu.support.zuora.api._
+import org.joda.time.{DateTimeZone, LocalDate}
+
+object ProductSubscriptionBuilders {
+
+  implicit class ContributionSubscriptionBuilder(val contribution: Contribution) extends ProductSubscriptionBuilder {
+    def build(config: ZuoraConfig): SubscriptionData = {
+      val contributionConfig = config.contributionConfig(contribution.billingPeriod)
+      buildProductSubscription(
+        contributionConfig.productRatePlanId,
+        List(
+          RatePlanChargeData(
+            ContributionRatePlanCharge(contributionConfig.productRatePlanChargeId, price = contribution.amount) //Pass the amount the user selected into Zuora
+          )
+        )
+      )
+    }
+  }
+
+  implicit class DigitalPackSubscriptionBuilder(val digitalPack: DigitalPack) extends ProductSubscriptionBuilder {
+    def build(config: ZuoraConfig, country: Country, maybePromoCode: Option[PromoCode], promotionService: PromotionService): SubscriptionData = {
+      val contractEffectiveDate = LocalDate.now(DateTimeZone.UTC)
+      val contractAcceptanceDate = contractEffectiveDate
+        .plusDays(config.digitalPack.defaultFreeTrialPeriod)
+        .plusDays(config.digitalPack.paymentGracePeriod)
+
+      val productRatePlanId = config.digitalPackRatePlan(digitalPack.billingPeriod)
+
+      val subscriptionData = buildProductSubscription(
+        productRatePlanId,
+        contractAcceptanceDate = contractAcceptanceDate,
+        contractEffectiveDate = contractEffectiveDate
+      )
+
+      maybePromoCode
```

Comment:
This is where promo codes are applied - the `PromotionService` updates the `subscriptionData` object to apply either a discount rate plan (for discounts) or to extend the contract acceptance date (for free trials). If the code is not applicable it will simply return the original `subscriptionData` unmodified. 

The `isRenewal` parameter is hard coded to false for now because we only deal with new subscriptions in support-workers. If this changes in future then we will need to pass in the correct value here.